vault/seal.go

```diff
@@ -267,7 +297,7 @@ func (d *defaultSeal) RecoveryConfig(ctx context.Context) (*SealConfig, error) {
 }
 
 func (d *defaultSeal) RecoveryKey(ctx context.Context) ([]byte, error) {
-	return nil, fmt.Errorf("recovery not supported")
+	return d.access.Wrapper.(*aead.ShamirWrapper).KeyBytes(ctx)
```

Comment:
It feels like this behaviour change is going to be confusing, esp in conjuction with VerifyRecoveryUnsealKey and SetRecoveryKey as defined here.  Maybe we should introduce a new method for this instead?